src/components/PharmacyList/PharmacyList.tsx

```diff
@@ -12,44 +13,83 @@ interface Pharmacy {
 }
 
 export function PharmacyList() {
-  const pharmacyList: Pharmacy[] = [
-    {
-      id: "1",
-      pharmacyName: "Stittsville IDA",
-      booking: true,
-      address: "250 Stittsville Main St. Stittsville, ON K2S 1S9",
-      phone: "613-835-3881",
-      website: "https://idapharmacy.com/",
-      lastUpdated: "N/a",
+  const { data } = useListVaccineAvailabilityApiV1VaccineAvailabilityGet({
+    queryParams: {
+      postalCode: "A1B 2G6",
     },
-    {
-      id: "2",
-      pharmacyName: "Costco Kanata",
-      booking: false,
-      address: "770 Silver Seven Road Ottawa, ON K2V 0A1",
-      phone: " 613-270-5500",
-      website: " https://www.costcopharmacy.ca/appointment",
-      lastUpdated: "N/a",
-    },
-    {
-      id: "3",
-      pharmacyName: "Example Pharmacy 1",
-      booking: false,
-      address: "123 Road Ottawa, ON K2V 0A1",
-      phone: " 613-111-1111",
-      website: " https://www.website.ca/appointment",
-      lastUpdated: "N/a",
-    },
-    {
-      id: "4",
-      pharmacyName: "Example Pharmacy 2",
-      booking: false,
-      address: "123 Road Ottawa, ON K2V 0A1",
-      phone: " 613-111-1111",
-      website: " https://www.website.ca/appointment",
-      lastUpdated: "N/a",
-    },
-  ];
+  });
+
+  const pharmacyList: Pharmacy[] =
+    data?.map((pharmacy) => {
+      const addressSegments: string[] = [];
+
+      const { address } = pharmacy.location;
+      if (address) {
+        if (address.line1) {
+          addressSegments.push(address.line1);
+        }
+
+        if (address.line2) {
+          addressSegments.push(address.line2);
+        }
+
+        if (address.city) {
+          addressSegments.push(address.city);
+        }
+
+        addressSegments.push(address.province);
+        addressSegments.push(address.postcode);
+      }
+
+      return {
+        id: pharmacy.id,
+        pharmacyName: pharmacy.location.name,
+        booking: true,
+        address: addressSegments.join(" "),
+        lastUpdated: "N/A",
+        phone: pharmacy.location.phone || "",
+        website: pharmacy.location.url || "",
+      };
+    }) || [];
```

Comment:
What do you think about not doing these prop transformations here and just passing the vaccine availability entry as a whole? The transformations here seem extraneous and for something like the address segments, that could just be handled in JSX? I think it may keep the types more simple since we don't need the `Pharmacy` interface?

Reply:
@johnhok I try to separate components that know about things external to themselves, and components that don't. It's basically the "smart/dumb" or "container/presentational" philosophy. As such, I don't want o give the entry directly to the pharmacy component as it is an API type, and then we have coupled a presentational component to our API. There is also more information in there than the component needs - it is better to be explicit about what exactly our component requires.

I did use the address segments to ensure spaces are added correctly - yes, it is probably overkill but I am OCD about my markup :) 

I did take your advice and dropped the pre-existing `Pharmacy` interface. It is better to infer props from the component you are pushing props into and supply them.

I've also re-named `PharmacyContainer` to `PharmacyCard` as "container" has a loaded meaning in React/Redux - it is a component that is aware of external state, which the component definitely should not.

Reply:
The new `PharmacyProps` typing is great. And those are great points. It would also make unit testing these components much easier without having to mock the whole API type.